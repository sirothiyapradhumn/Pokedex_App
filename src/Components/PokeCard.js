import React from 'react'
import {Link} from 'react-router-dom'

function PokeCard({row, data}) {
  return (
    <div class="card">
        <img src={row.img} alt={row.name}/>
        <div class="container">
            <h4><b>{row.num}</b></h4> 
            <p>{row.name}</p> 
            <p>{row.type.map((typ)=>typ+"  ")}</p> 
            <p>{row.weaknesses.map((weak)=>weak+"  ")}</p> 
            <p><button variant="outlined" onClick={()=>data(row)}>
            <Link to="/showdetails" style={{textDecoration:"none"}}>Show Details</Link>
                </button></p> 
        </div>
    </div>
  )
}

export default PokeCard