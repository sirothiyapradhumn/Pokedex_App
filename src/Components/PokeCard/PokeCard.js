import React from 'react'
import {Link} from 'react-router-dom'

function PokeCard({row, data}) {
  return (
    <div className="card" >
        <img src={row.img} alt={row.name}/>
        <div className="container">
            <h4><b>#{row.num}</b></h4> 
            <p className='pokeName'><b>{row.name}</b></p> 
            <p className='pokeType'>{row.type.map(type=> type+" ")}</p> 
            <button variant="outlined" onClick={()=>data(row)}>
            <Link to="/showdetails" style={{textDecoration:"none"}}>Show Details</Link>
            </button>
        </div>
    </div>
  )
}

export default PokeCard