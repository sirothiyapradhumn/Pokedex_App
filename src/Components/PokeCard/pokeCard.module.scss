.card {
    box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2);
    transition: 0.6s;
    width: 17vw;
    border-radius: 20px;
    background-color: #c5c6c7;
    margin: 1rem;
    img {
      border-radius: 41px 41px 41px 41px;
      width: 100%;
    }
  }
  
  .card:hover {
    // box-shadow: 0 16px 24px 0 rgba(28, 27, 27, 0.934);
    box-shadow: 0 20px 55px 0 #2196F3;
  }
  
  .container {
    padding: 1rem;
    h4{
      color: #434547b9;
      margin: 0%;
    }
    .pokeName{
      margin: 0%;
    }
    .pokeType{
      margin: 0%;
      padding-bottom: 1em;
    }
    button{
      border-radius: 28px;
      font-family: Arial;
      color: #ffffff;
      font-size: 13px;
      background: #0b0c10;
      padding: 10px 20px 10px 20px;
      text-decoration: "none";
    }
    button:hover {
      background: #9fa1a3;
      text-decoration: none;
    }
  }