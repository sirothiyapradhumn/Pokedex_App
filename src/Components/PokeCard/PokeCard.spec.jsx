import { shallow } from "enzyme";
import PokeCard from "./PokeCard";

describe("Pokecard component", () => {
  let mockrow = {
    id: 1,
    num: "001",
    name: "Bulbasaur",
    img: "http://www.serebii.net/pokemongo/pokemon/001.png",
    type: ["Grass", "Poison"],
    height: "0.71 m",
    weight: "6.9 kg",
    candy: "Bulbasaur Candy",
    candy_count: 25,
    egg: "2 km",
    spawn_chance: 0.69,
    avg_spawns: 69,
    spawn_time: "20:00",
    multipliers: [1.58],
    weaknesses: ["Fire", "Ice", "Flying", "Psychic"],
    next_evolution: [
      {
        num: "002",
        name: "Ivysaur",
      },
      {
        num: "003",
        name: "Venusaur",
      },
    ],
  };

  function mockFn(){
    //console.log(row);
  }

  it(" snapshot for mock row", () => {
    let wrapper = shallow(<PokeCard row={mockrow} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot for button click',()=>{
    let wrapper = shallow(<PokeCard row={mockrow} data={mockFn} />);
    let button = wrapper.find('button');
    button.simulate('click');
    expect(button).toMatchSnapshot();
  });

});
