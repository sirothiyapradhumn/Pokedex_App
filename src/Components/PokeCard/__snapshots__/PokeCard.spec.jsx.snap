// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokecard component  snapshot for mock row 1`] = `
<div
  className="card"
>
  <img
    alt="Bulbasaur"
    src="http://www.serebii.net/pokemongo/pokemon/001.png"
  />
  <div
    className="container"
  >
    <h4>
      <b>
        #
        001
      </b>
    </h4>
    <p
      className="pokeName"
    >
      <b>
        Bulbasaur
      </b>
    </p>
    <p
      className="pokeType"
    >
      Grass 
      Poison 
    </p>
    <button
      onClick={[Function]}
      variant="outlined"
    >
      <Link
        style={
          Object {
            "textDecoration": "none",
          }
        }
        to="/showdetails"
      >
        Show Details
      </Link>
    </button>
  </div>
</div>
`;

exports[`Pokecard component snapshot for button click 1`] = `
<button
  onClick={[Function]}
  variant="outlined"
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/showdetails"
  >
    Show Details
  </Link>
</button>
`;
